src/renderer/components/blocks/layout/IssuedCarbonYearlyChart.tsx

```diff
@@ -0,0 +1,117 @@
+  const groupedData: Record<string, Record<string, number>> = {};
+  data.forEach((item) => {
+    const isUnitStatusMatch = item.unitStatus === unitStatus;
+    const isUnitTypeMatch = unitType === 'all' || item.unitType === unitType;
+    if ((!unitStatus || isUnitStatusMatch) && isUnitTypeMatch) {
+      const { vintageYear, unitType, tonsCo2 } = item;
+      if (vintageYear && !groupedData[vintageYear]) {
+        groupedData[vintageYear] = {};
+      }
+      if (vintageYear && unitType && !groupedData[vintageYear][unitType]) {
+        groupedData[vintageYear][unitType] = 0;
+      }
+      if (vintageYear && unitType && tonsCo2) {
+        groupedData[vintageYear][unitType] += tonsCo2;
+      }
+    }
+  });
+
+  return groupedData;
+};
+
+const unitStatusOptions = [
+  { label: 'Retired', value: 'Retired' },
+  { label: 'Held', value: 'Held' },
+  { label: 'Cancelled', value: 'Cancelled' },
+];
+
+const IssuedCarbonYearlyChart = () => {
+  const [vintageYearRangeStart] = useQueryParamState('vintageYearRangeStart', `${new Date().getFullYear() - 9}`);
+  const [vintageYearRangeEnd] = useQueryParamState('vintageYearRangeEnd', `${new Date().getFullYear()}`);
+  const [unitStatus, setUnitStatus] = useQueryParamState('issuedCarbonUnitStatus', undefined);
+  const [unitType] = useQueryParamState('unitType', 'all');
+
+  const {
+    data: carbonCreditByStatusData,
+    isLoading: carbonCreditByStatusLoading,
+    error: carbonCreditByStatusError,
+  } = useGetTonsCo2Query({ unitType, vintageYearRangeStart, vintageYearRangeEnd, unitStatus });
+
+  if (carbonCreditByStatusLoading) {
+    return <SkeletonCard />;
+  }
+
+  if (carbonCreditByStatusError) {
+    return (
+      <p className="capitalize">
+        <FormattedMessage id={'unable-to-load-contents'} />
+      </p>
+    );
+  }
+
+  if (!carbonCreditByStatusData) {
+    return (
+      <p className="capitalize">
+        <FormattedMessage id={'no-records-found'} />
+      </p>
+    );
+  }
+
+  const lastTenYears = generateYearsRange(10)
+    .map((year) => year.label)
+    .reverse();
+  const carbonData = processCarbonDataByYearAndType(carbonCreditByStatusData.data, unitStatus, unitType);
+
+  const unitTypes = Array.from(
+    new Set(carbonCreditByStatusData.data.map((item) => item.unitType).filter((item) => Boolean(item))),
+  ) as string[];
+
+  const datasets = unitTypes.map((type) => ({
+    label: type,
+    data: lastTenYears.map((year) => carbonData[year]?.[type] || 0),
+  }));
+
+  const chartData2 = createChartDataWithMultipleDatasets(lastTenYears, datasets);
+
+  return (
+    <Card>
+      <div className="grid justify-end">
+        <Select
+          name="status"
+          options={unitStatusOptions}
+          initialValue={unitStatus}
+          onChange={(value) => {
+            setUnitStatus(String(value));
+          }}
+        />
+      </div>
+      <BarChart
```

Comment:
if the chart has no data to display, it it should display a message in the middle of its canvas saying that there is no data to display, the message should be translatable.

Reply:
when i ran the project, based on the behavior of the IssuedCarbonYearlyChart this still needs to be implemented